scipy/sparse/_base.py

```diff
@@ -1177,6 +1196,12 @@ def _is_integral(dtype):
         inter_dtype = np.float64 if integral else res_dtype
         inter_self = self.astype(inter_dtype)
 
+        if self.ndim == 1:
+            if axis not in (None, -1, 0):
+                raise ValueError("axis must be None, -1 or 0")
+            res = inter_self / np.array(self.shape[-1])
```

Comment:
Why is the `np.array()` call needed here?

Reply:
`np.array` is used for 2d, so I kept it in 1d. But removing it in both makes more sense and passes all tests.

I have made a commit to put in all these suggestions and also remove the extra `np.array` from the 2d code 3-4 lines down from this section. Remaining questions about these 3 suggestions are:
- should we raise error in `getcol` for 1d?
- should we remove the `np.array` from the 2d code?